src/bootstrap/config.rs

```diff
@@ -1910,11 +1939,28 @@ impl Config {
 
         // Look for a version to compare to based on the current commit.
         // Only commits merged by bors will have CI artifacts.
+
+        // If we are running in CI, the current commit will never have artifacts already built.
+        let tip_commit = if CiEnv::is_ci() { "HEAD^" } else { "HEAD" };
+        // NOTE: we intentionally don't filter compiler/ and library/ here, even though we ideally
+        // would to avoid unnecessary downloads. The reason for that is complicated:
+        // - When bors creates a merge commit for a rollup, there are two levels of indirection:
+        //   1. The merge by bors ("Auto merge of...")
+        //   2. The merge from each individual commit in the rollup ("Rollup merge of ...")
+        // - Because of the indirection, `git log --author=bors compiler/` *only* includes merges that weren't from a rollup.
```

Comment:
```suggestion
        // - `git log --author=bors compiler/` *only* includes the "Auto merge... " commits, including for rollups that touched the compiler/ directory.
```

* "rollup merge" is excluded _not_ because of indirection (AFAICT) but because the author is not bors. I think that's semi-reliable; those commits are currently created via GitHub API by the person clicking the "Create a rollup" button on bors.r-l.o. That might change with future bors iterations, certainly if we have automatic rollup creation of some kind.
* it *does* include the "Auto merge ..." commits of rollups in local testing, so the previous comment seems misleading.

Reply:
this comment is gone now - happy to answer questions on https://github.com/rust-lang/rust/issues/113250